### ISSUE TITLE
error: undefined local variable or method `host' for #<TerraformLandscape::TerraformPlan:0x00007ff6dea84530>.

### ISSUE THREAD

**Issue by andrewhertog** [coinbase/terraform-landscape#62]
Getting this error now with recent release of terraform aws provider version 1.33.0.

```
error: undefined local variable or method `host' for #<TerraformLandscape::TerraformPlan:0x00007ff6dea84530>.
```

Please see https://github.com/terraform-providers/terraform-provider-aws/issues/5661 for expanded details and stack trace

**Comment by jianyuan** [comment]
It's probably due to the fact that the new [fixed-response config](https://github.com/terraform-providers/terraform-provider-aws/pull/5430) contains valid Ruby string interpolation variables as Terraform values which are `eval`ed by terraform-landscape.

- `#{protocol}`
- `#{host}`
- `#{port}`
- `#{path}`
- `#{query}`

Culprits:

- https://github.com/coinbase/terraform-landscape/blob/master/lib/terraform_landscape/terraform_plan.rb#L181
- https://github.com/coinbase/terraform-landscape/blob/master/lib/terraform_landscape/terraform_plan.rb#L220

I do question the use of `eval`s as it poses a security concern.

**Comment by sds** [comment]
Totally agree about the use of `eval`. You can see that I explicitly ignore the security warning.

This was a quick solution to a problem, and it's clear we've outgrown it. The proper solution is to create a Treetop grammar to parse the hashrocket-style attribute declarations.

Would love help with this!

**Comment by robacarp** [comment]
Given that this isn't security software and therefore doesn't need to deal with malicious input data in the same way a search engine does, a full grammar is probably overkill. Could this be kicked down the road by simply running a quick and dirty escape on the input string, converting `#{` into `${`?

**Comment by sds** [comment]
Fixed in #73
